lib/ht_sip_validator/validator_config.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module HathiTrust # rubocop:disable Style/ClassAndModuleChildren
+  # Representation of configuration for a single Validator
+  class ValidatorConfig
+    attr_reader :prerequisites
+    attr_reader :validator_class
+
+    def initialize(validator_config)
+      @prerequisites = parse_prerequisites(validator_config).map do |prereq|
+        Validator.const_get(prereq)
+      end
+
+      @validator_class = Validator.const_get(parse_validator_class(validator_config))
+    end
+
+    private
+
+    def no_prereqs?(validator_config)
+      validator_config.is_a?(String)
+    end
+
+    def one_prerequisite?(validator_config)
+      validator_config.is_a?(Hash) && (validator_config.size == 1) &&
+        validator_config.values.first.is_a?(String)
+    end
+
+    def multi_prerequisites?(validator_config)
+      validator_config.is_a?(Hash) && (validator_config.size == 1) &&
+        validator_config.values.first.is_a?(Array) &&
+        validator_config.values.first.all? {|v| v.is_a?(String) }
+    end
+
+    def parse_prerequisites(validator_config)
```

Comment:
This may be needlessly complicated to support all these cases:
```yaml
- ValidatorWithNoDependencies
- ValidatorWithOneDependency: ValidatorWithNoDependencies
- ValidatorWithMultipleDependencies: [ValidatorOne, ValidatorTwo]
```

I could perhaps simplify this by either requiring the incoming yaml to always be in the following form, or by transforming the data to this form rather than checking for all options both here and in parse_validator_class
```yaml
- ValidatorWithNoDependencies: []
- ValidatorWithOneDependency: [ValidatorWithNoDependencies]
- ValidatorWithMultipleDependencies: [ValidatorOne, ValidatorTwo]
```